### PR TITLE
Improve Configuration Documentation

### LINKED ISSUE #227: Improve Configuration Documentation

### PR DESCRIPTION
**Pull request by curran**
This PR is about improving the documentation for configuration.

Closes #227

### CODE REVIEW

**Comment by curran** [comment]
@seemantk This is almost ready. Need to stop for today, but feel free to have a look at what's here, in terms of structure.

**Comment by curran** [comment]
@seemantk This is ready for review.

File: README.md [hunk 1]
```diff
@@ -4,15 +4,18 @@ An interactive visualization of the [Campaign Finance Institute](http://www.cfin
...
+[![image](https://cloud.githubusercontent.com/assets/68416/22932623/8e465200-f2ee-11e6-8c06-e9040a6476ae.png)](https://cfinst.github.io/)
 
-## Licensing
+## Table of Contents
 
-The source code is released under the GPL3 license ([LICENSE](LICENSE)).
+ * [Development Environment](#development-environment)
+ * [Configuration](#configuration)
+ * [Updating Data](#updating-data)
+ * [Licensing](#licensing)
```
Comment (seemantk) [inline README.md]: Nice ToC!

File: README.md [hunk 2]
```diff
@@ -29,7 +32,217 @@ jekyll serve --watch
...
+│   ├── about.md
+│   └── howto.md
+└── public-financing
+    ├── about.md
+    └── howto.md
+_sass
+├── _tips.scss
+└── _vis.scss
+```
+
```
Comment (seemantk) [inline README.md]: Gorgeous!

File: README.md [hunk 3]
```diff
@@ -29,7 +32,217 @@ jekyll serve --watch
...
+ * `scale` The scale entries.
+   * If `type` is `threshold`, each scale entry defines an interval of values, and has the following parameters:
+     * `min` The minimum value for this interval.
+     * `max` The maximum value for this interval.
+     * `label` The text to display in the legend and tooltips for this interval.
+     * `color` The name of the color, referencing named colors defined in `_data/colors.yml`.
+   * If `type` is `ordinal`, each scale entry has the following parameters:
+     * `label` The text to display in the legend and tooltips for this interval. This must match with data values from the database.
+     * `color` The name of the color, referencing named colors defined in `_data/colors.yml`.
+
```
Comment (seemantk) [inline README.md]: This is incredibly thorough. Nice work here!

File: README.md [hunk 4]
```diff
@@ -29,7 +32,217 @@ jekyll serve --watch
...
+ * `layout` The Jekyll layout to use for the modal HTML structure.
+ * `modal` The unique identifier for this modal. Please do not change this without updating all references to it.
+ * `title` The text displayed as the title of this modal.
+
+```
+_modals/:section/howto.md
+```
+
+This file specifies the content of the "Using this page" modal dialog for each section. Its Markdown body text is used as the content of the dialog, passed through a Markdown parser. Its YAML frontmatter defines the same parameters as in `about.md`.
+
```
Comment (seemantk) [inline README.md]: Brilliant stuff

**Pull request merged**